### PR TITLE
[Pages] _redirects proxying functionality

### PR DESCRIPTION
**Pull request by Skye-31**
To be merged after https://github.com/cloudflare/workers-sdk/pull/2708

Pages _redirects will now support proxying, for example `/users/:id /users/id 200`, where any request to `/users/*` is sent to `/users/id.html` internally, while the URL in the browser remains the same. This allows for client-side code to then dynamically render the page using the URL, without any Workers/Functions needed

### CODE REVIEW

**Review by WalshyDev — APPROVED** [review]

**Comment by GregBrimble** [comment]
We don't flatten redirects (i.e. if there is a rule for `/a /b 200` and `/b /c 200`, we won't automatically serve `/c` on `/a`). This is true for all of our redirects — not just proxying — we also won't redirect from `/a` to `/c` if those were 302s. However, it seems like this feature might raise the question for people, so can we add a note to this docs page somewhere that outlines our behavior? Maybe just after "If there are multiple redirects for the same source path, the topmost redirect is applied."?

Can we also add a note somewhere that duplicate content has the potential to harm SEO and include an example `_headers` rule which adds a canonical reference to the original page? This won't be relevant to every application which uses proxying, but it's probably worth letting people know if they weren't already aware.

Something like:

> Be aware that proxying pages can have an adverse effect on SEO. Search engines often penalize websites that serve duplicate content, so consider adding a `Link` HTTP header which informs search engines of the canonical source of content.
> 
> For example, if you have added `/about/faq/* /about/faqs 200` to your `_redirects` file, you will likely also want to add the following to your `_headers` file:
> ```txt
> /about/faq/*
>   Link: </about/faqs>; rel="canonical"
> ```

**Comment by Skye-31** [comment]
Done 🙂

File: content/pages/platform/redirects.md [hunk 1]
```diff
@@ -87,6 +87,26 @@ A placeholder can be defined with `:placeholder_name`. A colon indicates the sta
...
+```
+
+Be aware that proxying pages can have an adverse effect on SEO. Search engines often penalize websites that serve duplicate content, so consider adding a Link HTTP header which informs search engines of the canonical source of content.
+
+For example, if you have added `/about/faq/* /about/faqs 200` to your _redirects file, you will likely also want to add the following to your _headers file:
+
+```
+/about/faq/*
+  Link: </about/faqs>; rel="canonical"
+```
```
Comment (GregBrimble) [inline content/pages/platform/redirects.md]:
````suggestion
{{<Aside heading="Proxying's impact on SEO">}}
Be aware that proxying pages can have an adverse effect on search engine optimization (SEO). Search engines often penalize websites that serve duplicate content, so consider adding a `Link` HTTP header which informs search engines of the canonical source of content.

For example, if you have added `/about/faq/* /about/faqs 200` to your `_redirects` file, you will likely also want to add the following to your `_headers` file:

```txt
/about/faq/*
  Link: </about/faqs>; rel="canonical"
```
{{</Aside>}}
````

Maybe nice as an aside?

File: content/pages/platform/redirects.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ Cloudflare currently offers limited support for advanced redirects. More support
 | Placeholders                        | Yes     | `/blog/:year/:month/:date/:slug /news/:year/:month/:date/:slug` | See [Placeholders](#placeholders)                                                                 |
```
Comment (deadlypants1973) [inline content/pages/platform/redirects.md]:
```suggestion
| Placeholders                        | Yes     | `/blog/:year/:month/:date/:slug /news/:year/:month/:date/:slug` | Refer to [Placeholders](#placeholders).                                                                |
```

File: content/pages/platform/redirects.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ Cloudflare currently offers limited support for advanced redirects. More support
...
 | Query Parameters                    | No      | `/shop id=:id /blog/:id 301`                                    |                                                                                                   |
 | Force                               | Yes     | `/pagethatexists /otherpage`                                    | Redirects are always followed, regardless of whether or not an asset matches the incoming request. |
-| Proxying                            | No      | `/blog/* https://blog.my.domain/:splat 200`                     |                                                                                                   |
+| Proxying                            | Yes     | `/blog/* /news/:splat 200`                                      | Refer to [Proxying](#proxying)                                                                    |
```
Comment (deadlypants1973) [inline content/pages/platform/redirects.md]:
```suggestion
| Proxying                            | Yes     | `/blog/* /news/:splat 200`                                      | Refer to [Proxying](#proxying).                                                                   |
```

File: content/pages/platform/redirects.md [hunk 4]
```diff
@@ -87,6 +87,28 @@ A placeholder can be defined with `:placeholder_name`. A colon indicates the sta
 
 Similarly, the matched value can be used in the redirect location with `:placeholder_name`.
 
+#### Proxying
+
+Proxying will only support relative URLs on your site - i.e. you can't proxy external domains.
```
Comment (deadlypants1973) [inline content/pages/platform/redirects.md]:
```suggestion
Proxying will only support relative URLs on your site. You cannot proxy external domains.
```

File: content/pages/platform/redirects.md [hunk 5]
```diff
@@ -87,6 +87,28 @@ A placeholder can be defined with `:placeholder_name`. A colon indicates the sta
...
+
+Only the first redirect will apply, for example, in the following scenario, a request to `/a` will render `/b`, and a request to `/b` will render `/c`, but `/a` will not render `/c`.
```
Comment (deadlypants1973) [inline content/pages/platform/redirects.md]:
```suggestion
Only the first redirect in your will apply. For example, in the following example, a request to `/a` will render `/b`, and a request to `/b` will render `/c`, but `/a` will not render `/c`.
```

File: content/pages/platform/redirects.md [hunk 6]
```diff
@@ -87,6 +87,28 @@ A placeholder can be defined with `:placeholder_name`. A colon indicates the sta
...
+
+```
+/a /b 200
+/b /c 200
+```
+
+{{<Aside heading="Proxying's impact on SEO">}}
+Be aware that proxying pages can have an adverse effect on search engine optimization (SEO). Search engines often penalize websites that serve duplicate content, so consider adding a `Link` HTTP header which informs search engines of the canonical source of content.
+
+For example, if you have added `/about/faq/* /about/faqs 200` to your `_redirects` file, you will likely also want to add the following to your `_headers` file:
```
Comment (deadlypants1973) [inline content/pages/platform/redirects.md]:
```suggestion
Be aware that proxying pages can have an adverse effect on search engine optimization (SEO). Search engines often penalize websites that serve duplicate content. Consider adding a `Link` HTTP header which informs search engines of the canonical source of content.

For example, if you have added `/about/faq/* /about/faqs 200` to your `_redirects` file, you may want to add the following to your `_headers` file:
```

**Review by deadlypants1973 — APPROVED** [review]
PCX review

File: content/pages/platform/redirects.md (hunk 5 above)
Comment (Skye-31) [reply content/pages/platform/redirects.md]: @deadlypants1973 This sentence doesn't make sense anymore 😅

**Pull request merged**